tests/scripts/components-compiler.sh

```diff
@@ -0,0 +1,133 @@
+# components-compiler.sh
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+# This file contains test components that are executed by all.sh
+
+################################################################
+#### Compiler Testing
+################################################################
+
+support_build_tf_psa_crypto_tfm_armcc () {
+    support_build_tf_psa_crypto_armcc
+}
+
+component_build_tf_psa_crypto_tfm_armcc () {
+    # test the TF-M configuration can build cleanly with various warning flags enabled
+    cp configs/ext/crypto_config_profile_medium.h "$CRYPTO_CONFIG_H"
+
+    msg "build: TF-M config, armclang armv7-m thumb2"
+    helper_armc6_build_test "--target=arm-arm-none-eabi -mcpu=cortex-m0 -mthumb -Os -std=c99 -Werror -Wall -Wextra -Wwrite-strings -Wpointer-arith -Wimplicit-fallthrough -Wshadow -Wvla -Wformat=2 -Wno-format-nonliteral -Wshadow -Wasm-operand-widths -Wunused -I../framework/tests/include/spe"
+}
+
+test_build_opt () {
+    info=$1 cc=$2; shift 2
+    $cc --version
+    for opt in "$@"; do
+          cd $OUT_OF_SOURCE_DIR
+          msg "build/test: $cc $opt, $info" # ~ 30s
+          cmake -DCMAKE_C_COMPILER="$cc" -DCMAKE_C_FLAGS="$opt -std=c99 -pedantic -Wall -Wextra -Werror" "$TF_PSA_CRYPTO_ROOT_DIR"
+          make
+          # We're confident enough in compilers to not run _all_ the tests,
+          # but at least run the unit tests. In particular, runs with
+          # optimizations use inline assembly whereas runs with -O0
+          # skip inline assembly.
+          make test # ~30s
+    done
+}
+
+# For FreeBSD we invoke the function by name so this condition is added
+# to disable the existing test_clang_opt function for linux.
+if [[ $(uname) != "Linux" ]]; then
+    component_test_tf_psa_crypto_clang_opt () {
+        scripts/config.py full
+        test_build_opt 'full config' clang -O0 -Os -O2
+    }
+fi
+
+component_test_tf_psa_crypto_clang_latest_opt () {
+    scripts/config.py full
+    test_build_opt 'full config' "$CLANG_LATEST" -O0 -Os -O2
+}
+
+support_test_tf_psa_crypto_clang_latest_opt () {
+    type "$CLANG_LATEST" >/dev/null 2>/dev/null
+}
+
+component_test_tf_psa_crypto_clang_earliest_opt () {
+    scripts/config.py full
+    test_build_opt 'full config' "$CLANG_EARLIEST" -O2
+}
+
+support_test_tf_psa_crypto_clang_earliest_opt () {
+    type "$CLANG_EARLIEST" >/dev/null 2>/dev/null
+}
+
+component_test_tf_psa_crypto_gcc_latest_opt () {
+    scripts/config.py full
+    test_build_opt 'full config' "$GCC_LATEST" -O0 -Os -O2
+}
+
+support_test_tf_psa_crypto_gcc_latest_opt () {
+    type "$GCC_LATEST" >/dev/null 2>/dev/null
+}
+
```

Comment:
Note that https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/272 adds a component to `components-compiler.sh` (for GCC 15) and the new component is desired in TF-PSA-Crypto too. Given that both PR appear to be close to completion, I think it would be best to wait until both are merged, then do a follow-up.

Reply:
https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/272 is now merged.

Reply:
Reminder that “[All components but component_build_mingw](https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/60)” now includes `component_test_gcc15_drivers_opt`.

Reply:
(Note that I'm not actively reviewing this, I just made a note of related work going on in parallel.)

Reply:
I've just created https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/304 to cover `component_test_gcc15_drivers_opt`.